internal/ingress/controller/parser/parser.go

```diff
@@ -497,10 +497,20 @@ func (p *Parser) parseIngressRules(
 func (p *Parser) fillOverrides(state KongState) error {
 	for i := 0; i < len(state.Services); i++ {
 		// Services
-		kongIngress, err := p.getKongIngressForService(
-			state.Services[i].Namespace, state.Services[i].Backend.ServiceName)
+		svc, err := p.store.GetService(
+			state.Services[i].Namespace,
+			state.Services[i].Backend.ServiceName)
+		var svcAnns map[string]string
 		if err == nil {
-			overrideService(&state.Services[i], kongIngress)
+			svcAnns = svc.Annotations
+		}
```

Comment:
Always handle the error. Either log it or return it up. In this case, we want to log it, because if there is an error retrieving one service, we still want to process other services.